src/uniforms/value.rs

```diff
@@ -962,3 +964,17 @@ impl AsUniformValue for (u64, u64, u64, u64) {
 }
 
 impl_uniform_block_basic!((u64, u64, u64, u64), UniformType::UnsignedInt64Vec4);
+
+impl<'a> AsUniformValue for (ShaderStage, &'a str) {
+    #[inline]
+    fn as_uniform_value(&self) -> UniformValue {
+        UniformValue::Subroutine(self.0, self.1)
+    }
+}
+
+impl<'a> AsUniformValue for (&'a str, ShaderStage) {
```

Comment:
I'll stick to the `(&str, ShaderStage)` variant, which I think is more intuitive.